src/Symfony/Component/Security/Http/Authenticator/AuthenticatorInterface.php

```diff
@@ -47,8 +48,10 @@ public function supports(Request $request): ?bool;
      * a UserNotFoundException when the user cannot be found).
      *
      * @throws AuthenticationException
+     *
+     * @return Passport
      */
-    public function authenticate(Request $request): PassportInterface;
+    public function authenticate(Request $request); /*: Passport;*/
```

Comment:
Well, maybe we should keep the existing return type while documenting the new one, instead of widening the explicit return type now (the new one is only documented anyway).

Reply:
:+1: if the `DebugClassLoader` still correctly reports the deprecation.

Reply:
Keeping the return type would forbid upgrading userland authenticators on PHP versions prior to 7.4 https://3v4l.org/LcLRN

Reply:
I don't see much issue in requiring a user to be on PHP 7.4 to fix a deprecation (i.e. it's not breaking support for older versions, it's just triggering a deprecation if you stay on `: PassportInterface`). They would need PHP 8 anyways if they want to upgrade to Symfony 6 and even PHP 7.4 will be eom when we release Symfony 5.4.

Reply:
Pushing our users to upgrade their PHP versions has never been our take, at least not by making symfony upgrades harder. Moreover, that would hardly break our CI.